AdobeStockImage/Model/SerializeImage.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Model;
+
+use Magento\AdobeStockImageApi\Api\SerializeImageInterface;
+use Magento\Framework\Api\AttributeInterface;
+use Magento\Framework\Api\Search\Document;
+use Magento\Framework\Exception\SerializationException;
+
+/**
+ * Class used to serialize a Document object
+ */
+class SerializeImage implements SerializeImageInterface
+{
+    /**
+     * Serializes a Document object
+     *
+     * @param Document $image
+     * @return array
+     * @throws SerializationException
+     */
+    public function execute(Document $image): array
+    {
+        $data = [];
+        try {
+            /** @var AttributeInterface $attribute */
+            foreach ($image->getCustomAttributes() as $attribute) {
+                if ($attribute->getAttributeCode() === 'thumbnail_240_url') {
+                    $data['thumbnail_url'] = $attribute->getValue();
+                    continue;
+                }
+                $data[$attribute->getAttributeCode()] = $attribute->getValue();
+            }
+            return $data;
+        } catch (\Exception $exception) {
+            throw new SerializationException(
+                __(
+                    'An error occurred during image serialization: %error',
+                    ['error' => $exception->getMessage()]
+                )
+            );
```

Comment:
```suggestion
            throw new SerializationException( __('An error occurred during image serialization.));
```
According to the latest changes in the ASI an exception should be thrown but not with a specified error message. We specified exception message previously but this appeared confusing to the end-user. An exception should be logged but not thrown. In that particular case, log message is not required because of the log present in the method which calls this one.